### ISSUE TITLE
ImportError: cannot import name 'contextfilter' from 'jinja2'

### ISSUE THREAD

**Issue by eimrek** [aiidalab/aiidalab-launch#165]
Hi! After doing a fresh set-up, getting a "500 : Internal Server Error" in the jupyter notebook, when opening the URL.

Docker logs:

```
[E 15:42:43.375 NotebookApp] Uncaught exception GET /apps/apps/home/start.ipynb?token=ecd4729ff2ec32c37df31fd8f8dd60948ad0186850ec1a86d5173d519309eea8 (172.17.0.1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='GET', uri='/apps/apps/home/start.ipynb?token=ecd4729ff2ec32c37df31fd8f8dd60948ad0186850ec1a86d5173d519309eea8', version='HTTP/1.1', remote_ip='172.17.0.1')
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.9/site-packages/tornado/web.py", line 1711, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/opt/conda/lib/python3.9/site-packages/tornado/web.py", line 3208, in wrapper
        return method(self, *args, **kwargs)
      File "/opt/conda/lib/python3.9/site-packages/appmode/server_extension.py", line 86, in get
        self.write(self.render_template('appmode.html', **render_kwargs))
      File "/opt/conda/lib/python3.9/site-packages/notebook/base/handlers.py", line 511, in render_template
        return template.render(**ns)
      File "/opt/conda/lib/python3.9/site-packages/jinja2/environment.py", line 1301, in render
        self.environment.handle_exception()
      File "/opt/conda/lib/python3.9/site-packages/jinja2/environment.py", line 936, in handle_exception
        raise rewrite_traceback_stack(source=source)
      File "/opt/conda/lib/python3.9/site-packages/appmode/appmode.html", line 1, in top-level template code
        {% extends "notebook.html" %}
      File "/opt/conda/lib/python3.9/site-packages/notebook/templates/notebook.html", line 1, in top-level template code
        {% extends "page.html" %}
      File "/opt/conda/lib/python3.9/site-packages/notebook/templates/page.html", line 154, in top-level template code
        {% block header %}
      File "/opt/conda/lib/python3.9/site-packages/notebook/templates/notebook.html", line 115, in block 'header'
        {% for exporter in get_frontend_exporters() %}
      File "/opt/conda/lib/python3.9/site-packages/notebook/notebook/handlers.py", line 41, in get_frontend_exporters
        exporter_class = get_exporter(name)
      File "/home/jovyan/.local/lib/python3.9/site-packages/nbconvert/exporters/base.py", line 98, in get_exporter
        return entrypoints.get_single('nbconvert.exporters', name).load()
      File "/opt/conda/lib/python3.9/site-packages/entrypoints.py", line 79, in load
        mod = import_module(self.module_name)
      File "/opt/conda/lib/python3.9/importlib/__init__.py", line 127, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
      File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
      File "<frozen importlib._bootstrap>", line 972, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
      File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
      File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
      File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 850, in exec_module
      File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
      File "/home/jovyan/.local/lib/python3.9/site-packages/nbconvert/__init__.py", line 4, in <module>
        from .exporters import *
      File "/home/jovyan/.local/lib/python3.9/site-packages/nbconvert/exporters/__init__.py", line 3, in <module>
        from .html import HTMLExporter
      File "/home/jovyan/.local/lib/python3.9/site-packages/nbconvert/exporters/html.py", line 12, in <module>
        from jinja2 import contextfilter
    ImportError: cannot import name 'contextfilter' from 'jinja2' (/opt/conda/lib/python3.9/site-packages/jinja2/__init__.py)
```

Seems to originate from an incompatibility of the explicitly pinned `nbconvert==5.6.1` and the `jinja2==3.1.2` that is installed by the conda feedstock (not sure how this works exactly).

**Comment by danielhollas** [comment]
Hi @eimrek. Can you please show us the configuration of your `aiidalab-launch` profiles?

```console
aiidalab-launch profiles show default
```

(subtitute `default` with the name of your profile in case you're not using the default)

> Seems to originate from an incompatibility of the explicitly pinned nbconvert==5.6.1 and the jinja2==3.1.2 that is installed by the conda feedstock (not sure how this works exactly).

Where do you see this pinned exactly?

**Comment by eimrek** [comment]
Thanks for the quick reply @danielhollas 

Here:

```
(base) kristjan@vxub:~$ aiidalab-launch profiles show default
port = 8888
default_apps = [ "aiidalab-widgets-base",]
system_user = "jovyan"
image = "aiidalab/full-stack:latest"
home_mount = "aiidalab_default_home"
extra_mounts = []
```

```
(base) kristjan@vxub:~$ docker ps
CONTAINER ID   IMAGE          COMMAND                  CREATED       STATUS                 PORTS                                       NAMES
2c902f4c896a   e7bff19f093a   "tini -g -- start-no…"   3 hours ago   Up 3 hours (healthy)   0.0.0.0:8888->8888/tcp, :::8888->8888/tcp   aiidalab_default
(base) kristjan@vxub:~$ docker exec -it 2c902f4c896a /bin/bash
(base) jovyan@2c902f4c896a:~$ pip freeze | grep nbconvert
nbconvert==5.6.1
(base) jovyan@2c902f4c896a:~$ pip freeze | grep jinja2
Jinja2 @ file:///home/conda/feedstock_root/build_artifacts/jinja2_1654302431367/work
```

True, what i said is probably not correct: the `nbconvert` maybe is not explicitly pinned but this is the version i got with `aiidalab-launch start`.

**Comment by danielhollas** [comment]
Hi @eimrek,

I think the issue is that we have not yet released a new stable version of aiidalab-widgets-base that works together with the aiiidalab/full-stack image. Can you please try modifying your profile with this?

```
default_apps = [ "aiidalab-widgets-base=2.0b1",]
```

**Comment by danielhollas** [comment]
Just to double check, did you reset the container? Once the container starts and default apps are installed, they do not get reinstalled until you reset (but watch out this will delete all the data!!)

aiidalab-launch reset

Or you can reinstall the app from the app store, that might be easier.

**Comment by eimrek** [comment]
nice! the reset helped! Thanks!

(But i guess this issue is still relevant until the `aiidalab-widgets-base` is released)

**Comment by danielhollas** [comment]
We've removed the `aiidalab-widgets-base` from the default apps in #166 so I am closing this. Obviously, the proper fix is to release the stable AWB 2.0 version, we should not be too far from that.
